std/numeric.d

```diff
@@ -2934,49 +2934,56 @@ Params:
 Returns:
     The greatest common divisor of the given arguments.
  */
-T gcd(T)(T a, T b)
-if (isIntegral!T)
+typeof(Unqual!(T).init % Unqual!(U).init) gcd(T, U)(T a, U b)
+if (isIntegral!T && isIntegral!U)
 {
-    static if (is(T == const) || is(T == immutable))
+    import std.math : abs;
+    static if (is(T == const) || is(T == immutable) || is(U == const) || is(U == immutable))
     {
-        return gcd!(Unqual!T)(a, b);
-    }
-    else version (DigitalMars)
-    {
-        static if (T.min < 0)
-        {
-            assert(a >= 0 && b >= 0);
-        }
-        while (b)
-        {
-            immutable t = b;
-            b = a % b;
-            a = t;
-        }
-        return a;
+        return gcd!(Unqual!T, Unqual!U)(a, b);
     }
     else
     {
+        // Make arguments unsigned.
+        static if (T.min < 0)
+            a = abs(a);
+        static if (U.min < 0)
+            b = abs(b);
+
+        // Special cases.
         if (a == 0)
             return b;
         if (b == 0)
             return a;
 
-        import core.bitop : bsf;
-        import std.algorithm.mutation : swap;
+        version (DigitalMars)
+        {
+            while (b)
+            {
+                immutable t = b;
+                b = a % b;
```

Comment:
`gcd(-15,cast(ubyte) 45)` breaks here with "Error: cannot implicitly convert expression `a % cast(int)b` of type `int` to `ubyte`"